src/utils/searchViewHelper.ts

```diff
@@ -0,0 +1,77 @@
+import { FormEvent, MouseEvent } from 'react';
+
+import store from '../common/store';
+import {
+  updateSearchValue,
+  updateSearchResults,
+  updateRepository,
+  setInputError,
+  clearInputError,
+} from '../reducers/SearchSlice';
+import {
+  fetchRecentRevisions,
+  fetchRevisionByID,
+  fetchRevisionsByAuthor,
+} from '../thunks/searchThunk';
+import type { Repository } from '../types/state';
+
+let timeout: NodeJS.Timeout;
+
+const searchByRevisionOrEmail = (
+  repository: Repository['name'],
+  search: string,
+) => {
+  const emailMatch = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  const longHashMatch = /\b[a-f0-9]{40}\b/;
+  const shortHashMatch = /\b[a-f0-9]{12}\b/;
+
+  if (emailMatch.test(search)) {
+    store.dispatch(fetchRevisionsByAuthor({ repository, search }));
+  } else if (longHashMatch.test(search) || shortHashMatch.test(search)) {
+    store.dispatch(fetchRevisionByID({ repository, search }));
+  } else {
+    store.dispatch(
+      setInputError(
+        'Search must be a 12- or 40-character hash, or email address',
+      ),
+    );
+  }
+};
+
+const handleChangeDropdown = (event: MouseEvent) => {
+  const repository = (event.target as Element).id as Repository['name'];
+
+  // Update state with selected repository
+  store.dispatch(updateRepository(repository));
+
+  // Fetch 10 most recent revisions when repository changes
+  store.dispatch(fetchRecentRevisions(repository));
+};
+
+const handleChangeSearch = (
+  event: FormEvent<HTMLInputElement | HTMLTextAreaElement>,
+) => {
+  const search = event.currentTarget.value;
+  const { repository } = store.getState().search;
+  store.dispatch(updateSearchValue(search));
+  store.dispatch(updateSearchResults([]));
+  store.dispatch(clearInputError());
```

Comment:
Julien suggested using useDispatch and to probably avoid using store directly, not sure if it applies here.

Reply:
I can address that in another PR. These function had already been using `store.dispatch` so that part isn't new